cmd/kubeadm/app/apis/kubeadm/v1beta4/types.go

```diff
@@ -181,8 +184,19 @@ type APIServer struct {
 
 // DNS defines the DNS addon that should be used in the cluster
 type DNS struct {
-	// ImageMeta allows to customize the image used for the DNS component
+	// ImageMeta allows to customize the image used for the DNS addon
 	ImageMeta `json:",inline"`
+
+	// Disabled specifies whether to disable this addon in the cluster
+	// +optional
+	Disabled bool `json:"disabled,omitempty"`
+}
+
+// Proxy defines the proxy addon that should be used in the cluster
+type Proxy struct {
+	// Disabled specifies whether to disable this addon in the cluster
+	// +optional
+	Disabled bool `json:"disabled,omitempty"`
```

Comment:
@chendave do you think we can face a problem where we need this to be a *bool?
i.e. we must know when the user set it explicitly to `false`?
(same for proxy)

i think, no.

Reply:
By default, it's false. There is a conflict if user manually specified it's as false, but the phase is intending to skip via the flags `--skip-phases`, I think the logic is already covered by `manageSkippedAddons`, so I am fine with it.